django/core/handlers/asgi.py

```diff
@@ -223,6 +227,17 @@ async def process_request(request, send):
                     pass
         body_file.close()
 
+        if response is not None:
+            # If response exists, we must ensure `close` gets called. Do not allow this
+            # to be cancelled
+            await asyncio.shield(sync_to_async(response.close, thread_sensitive=True)())
```

Comment:
If we handle the `RequestAborted` case, do we need to move this here? 

i.e. I see the failure to dispatch `request_finished` on the cancelled case, but not (yet in tests at least) otherwise 🤔